src/pep8ext_naming.py

```diff
@@ -54,8 +54,11 @@ def err(self, node, code: str, **kwargs):
         return lineno, col_offset + 1, f'{code} {code_str}', self
 
 
-def _ignored(name, ignore):
-    return any(fnmatchcase(name, i) for i in ignore)
+class NameSet(frozenset[str]):
+    """A set of names that are matched using fnmatchcase."""
+
+    def __contains__(self, item) -> bool:
+        return any(fnmatchcase(item, name) for name in self)
```

Comment:
If we have to iterate every time and the set ordering is non deterministic, why are we using a set here? Also, does it make sense to cache any of this for speed? I doubt it's much of a hit but if we cache the contains call would that help?

Reply:
It was originally a set to de-duplicate any items provided by the `--ignore-names` option. Because our check supports globbing (via `fnmatch.`), we otherwise don't benefit from the set, so we could represent is here as a list (after de-duplicating).

Reply:
Caching (via `functools.lru_cache`) makes effectively no difference on test suite runs. I'll look at a larger project, too, but I think that the number of candidate names are so wide that we won't get great cache behavior in any case.

I think one optimization could be to look at all of the ignore names when this set is constructed, and if none of them contain globs, then just use the set's `__contains__` implementation. (Or store glob ignores in their own list and handle them specially.) Globs are probably the uncommon case anyway.

Reply:
The glob case can probably be done some other time if we find we need it